From f1d1fb49df935db61459a9a9f5a1b8e778fc6baa Mon Sep 17 00:00:00 2001
From: Pete Bentley <prb@google.com>
Date: Fri, 4 Dec 2020 09:26:35 +0000
Subject: [PATCH 3/3] Reject non-ASCII hostnames and SANs.

Updates test expectations to match OkHostnameVerifier behaviour changes
in the same topic.

Bug: 171980069
Test: atest CtsLibcoreTestCases CtsLibcoreOkHttpTestCases
Change-Id: I7ccd2c6d4255d29112e9a6d58640316c364bf67a
(cherry picked from commit 650e71cc3acbc3f5af2ae312bc5ed71bb2d7c20b)
---
 .../harmony/tests/javax/net/ssl/HostnameVerifierTest.java   | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/harmony-tests/src/test/java/org/apache/harmony/tests/javax/net/ssl/HostnameVerifierTest.java b/harmony-tests/src/test/java/org/apache/harmony/tests/javax/net/ssl/HostnameVerifierTest.java
index e6978385e0..96384adba2 100644
--- a/harmony-tests/src/test/java/org/apache/harmony/tests/javax/net/ssl/HostnameVerifierTest.java
+++ b/harmony-tests/src/test/java/org/apache/harmony/tests/javax/net/ssl/HostnameVerifierTest.java
@@ -78,7 +78,7 @@ public class HostnameVerifierTest extends TestCase implements
         assertFalse(verifier.verify("a.foo.com", session));
         assertTrue(verifier.verify("bar.com", session));
         assertFalse(verifier.verify("a.bar.com", session));
-        // The certificate has this name in the altnames section, but Conscrypt drops
+        // The certificate has this name in the altnames section, but OkHostnameVerifier drops
         // any altnames that are improperly encoded according to RFC 5280, which requires
         // non-ASCII characters to be encoded in ASCII via Punycode.
         assertFalse(verifier.verify("\u82b1\u5b50.co.jp", session));
@@ -129,11 +129,11 @@ public class HostnameVerifierTest extends TestCase implements
         assertFalse(verifier.verify("a.b.foo.com", session));
         assertFalse(verifier.verify("bar.com", session));
         assertTrue(verifier.verify("www.bar.com", session));
-        assertTrue(verifier.verify("\u82b1\u5b50.bar.com", session));
         assertFalse(verifier.verify("a.b.bar.com", session));
-        // The certificate has this name in the altnames section, but Conscrypt drops
+        // The certificate has this name in the altnames section, but OkHostnameVerifier drops
         // any altnames that are improperly encoded according to RFC 5280, which requires
         // non-ASCII characters to be encoded in ASCII via Punycode.
+        assertFalse(verifier.verify("\u82b1\u5b50.bar.com", session));
         assertFalse(verifier.verify("\u82b1\u5b50.co.jp", session));
         assertFalse(verifier.verify("a.\u82b1\u5b50.co.jp", session));
     }
-- 
2.36.0

